app/src/play/kotlin/org/thoughtcrime/securesms/notifications/FirebaseTokenFetcher.kt

```diff
@@ -1,19 +1,28 @@
 package org.thoughtcrime.securesms.notifications
 
-import com.google.android.gms.tasks.Tasks
-import com.google.firebase.iid.FirebaseInstanceId
+import com.google.firebase.messaging.FirebaseMessaging
+
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.isActive
+import kotlinx.coroutines.tasks.await
 import kotlinx.coroutines.withContext
 import javax.inject.Inject
 import javax.inject.Singleton
 
 @Singleton
 class FirebaseTokenFetcher @Inject constructor(): TokenFetcher {
     override suspend fun fetch() = withContext(Dispatchers.IO) {
+        /*
+        // OG: Firebase 18.0.0 code:
         FirebaseInstanceId.getInstance().instanceId
             .also(Tasks::await)
             .takeIf { isActive } // don't 'complete' task if we were canceled
             ?.run { result?.token ?: throw exception!! }
+        */
```

Comment:
Have you tested this? Any reason to keep the commented code?

Reply:
Upgrading Firebase was a breaking change which required code adjustment. While I did my research and I've tested the new code and everything seems to work, I wasn't confident to completely remove it and say that "Yes, this code is a 100% legit replacement for the existing code" (although I believe it to be).

@mpretty-cyro Would you be able to skim these few lines and say whether you're happy the replacement is suitable?

Reply:
If push notifications are still working then it's probably fine

I don't have a good handle on the differences between the two Firebase versions (and am not really across the different Android async behaviours), but one minor difference I can see is that previously we were checking `isActive` after awaiting, but now we are doing it before - I'm just wondering whether this could lead to a situation where `isActive` gets set to `false` while the token is being retrieved and end up being `null` because process was cancelled (which would result in the exception being thrown)?

Reply:
> If push notifications are still working then it's probably fine
>  ...

Have modified the line to more closely match the original (await first, and *then* take if active!):
`FirebaseMessaging.getInstance().token.await().takeIf { isActive } ?: throw Exception("Firebase token is null")`